rfc/rfc021-vp_token-grant-type.md

```diff
@@ -0,0 +1,162 @@
+The scope parameter determines the set of credentials the authorization server expects.
+
+The "presentation_submission" parameter MUST be used, as defined in Presentation Exchange [PE], to indicate how the VP matches the requested Verifiable Credentials.
+The Presentation Submission MUST be encoded using the application/x-www-form-urlencoded content type.
+
+The following example demonstrates an access token request with a VP as an authorization grant (with extra line breaks for display purposes only):
+
+     POST /token.oauth2 HTTP/1.1
+     Host: as.example.com
+     Content-Type: application/x-www-form-urlencoded
+
+     grant_type=urn%3Aietf%3Aparams%3Aoauth%3Agrant-type%3Avp_token-bearer
+     &assertion=eyJhbGciOiJFUzI1NiIsImtpZCI6IjE2In0.
+     eyJpc3Mi[...omitted for brevity...].
+     J9l-ZhwP[...omitted for brevity...]
+     &presentation_submission= {..}
+
+## 3.1 Assertion Encoding
+
+As stated in the previous section, the assertion parameter can either be encoded in JSON or JWT.
+The authorization server determines the encoding by adding the correct parameters to the authorization server metadata [RFC8414]].
+The following parameter MUST be added:
+
+* `vp_formats`: An object defining the formats and proof types of Verifiable Presentations and Verifiable Credentials that a Verifier supports as stated by §9 of the OpenID for Verifiable Presentations specification [OIDC4VP].
+  (ldp_vp or jwt_vp)
+
+When processing the assertion the authorization server can detect the used format by checking the initial character.
+A JSON encoded assertion starts with a '{' character or '%7B' when urlencoded.
+
+## 4. Processing requirements
+
+In order to issue an access token response as described in OAuth 2.0 [RFC6749], the authorization server MUST validate the VP.
+The validation requirements are split into three paragraphs. The first paragraph describes the requirements that apply to all VP's.
+The second paragraph describes the requirements that apply to JWT encoded VP's. The third paragraph describes the requirements that apply to JSON encoded VP's.
+
+### 4.1 General requirements
+
+1. The authorization server MUST validate the Presentation Submission according to section 6 of the Presentation Exchange [PE] specification.
+2. The authorization server MUST validate the VP according to section 6.1 of the Presentation Exchange [PE] specification.
+3. The Presentation Submission MUST be an answer to the Presentation Definition that corresponds with the requested scope. 
+   How an authorization server determines the Presentation Definition based on scope is out of scope of this specification.
+   However, a request to the Presentation Definition endpoint MUST result to the same Presentation Definition as the Presentation Definition that is used to validate the VP.
+4. The Verifiable Presentation MUST be not be valid for more than 15 minutes.
```

Comment:
why? This totally depends on the use case?

Reply:
Not really, it's created to get an access token. Once you have the access token, the VP is no longer needed.

Reply:
> Most of our RFCs are a mix of technical concepts combined with profile-esque components though.

I don't think this is necesarrily bad (for us), but we shoukd probably clearly split things (like you suggested putting this instance in security considerations).